omnigan/optim.py

```diff
@@ -33,24 +33,41 @@ def get_scheduler(optimizer, hyperparameters, iterations=-1):
     return scheduler
 
 
-def get_optimizer(net, opt_conf, iterations=-1):
+def get_optimizer(net, opt_conf, tasks=None, iterations=-1):
     """Returns a tuple (optimizer, scheduler) according to opt_conf which
     should come from the trainer's opts as: trainer.opts.<model>.opt
 
     Args:
         net (nn.Module): Network to update
         opt_conf (addict.Dict): optimizer and scheduler options
+        tasks: list of tasks
         iterations (int, optional): Last epoch number. Defaults to -1, meaning
             start with base lr.
 
     Returns:
         Tuple: (torch.Optimizer, torch._LRScheduler)
     """
     opt = scheduler = None
+    if tasks is None:
+        lr = opt_conf.lr
+        params = net.parameters()
+    else:
+        lr = opt_conf.lr["m"]
+        params = list()
+        for task in tasks:
+            # Parameters for encoder
+            if task == "m":
+                parameters = net.encoder.parameters()
+            # Parameters for decoders
+            if task == "p":
+                parameters = net.painter.parameters()
+            else:
+                parameters = net.decoders[task].parameters()
```

Comment:
for task `m` the encoder's parameters will be overwritten

Reply:
hmmm no, why would they?